rest/src/main/scala/io/udash/rest/RestDataCompanion.scala

```diff
@@ -109,3 +100,24 @@ abstract class RestDataWrapperCompanion[Wrapped, T](implicit
     (resolver: SchemaResolver, schemaTransform: RestSchema[_] => RestSchema[_]) =>
       wrappedResponses.responses(resolver, ws => schemaTransform(nameAndAdjusters.restSchema(ws)))
 }
+
+/**
+  * Default base class for companion objects of wrappers over other data types (i.e. case classes with single field)
+  * that uses default [[DefaultRestImplicits]] implicits.
+  *
+  * Using this base companion class makes the wrapper class effectively "transparent", i.e. as if it was annotated with
+  * [[com.avsystem.commons.serialization.transparent transparent]] annotation.
+  *
+  * @see [[BaseRestDataWrapperCompanion]]
+  * @example
+  * {{{
+  *   case class UserId(id: String) extends AnyVal
+  *   object UserId extends RestDataWrapperCompanion[String, UserId]
+  * }}}
+  */
+abstract class RestDataWrapperCompanion[Wrapped, T](implicit
+  instances: MacroInstances[DefaultRestImplicits, () => NameAndAdjusters[T]]
+) extends TransparentWrapperCompanion[Wrapped, T] with BaseRestDataWrapperCompanion[Wrapped, T] {
+  final override protected lazy val nameAndAdjusters: NameAndAdjusters[T] = instances(DefaultRestImplicits, this).apply()
```

Comment:
If we need to cache it (I don't remember the internals of `MacroInstances`), we should add a comment here.